cockroachcloud/cloud-api.md

```diff
@@ -390,6 +390,46 @@ Where:
 
 If the `DELETE` request was successful the client will not receive a response payload.
 
+<a id="cloud-audit-logs"></a>
+
+## Export {{ site.data.products.db }} audit logs
+
+{% include feature-phases/preview-opt-in.md %}
+
+To export audit logs for activities and events related to your {{ site.data.products.db }} organization, send a `GET` request to the `/v1/auditlogevents` endpoint. The service account associated with the secret key must have `ADMIN` [permission](console-access-management.html#service-accounts).
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+curl --request GET \
+  --url 'https://cockroachlabs.cloud/api/v1/auditlogevents?StartingFrom={timestamp},SortOrder={sort_order},limit={limit}' \
+  --header 'Authorization: Bearer {secret_key}'
+~~~
+
+Where:
+
+  - `{timestamp}` indicates the first log entry to fetch. If unspecified, `{timestamp}` is set to the beginning of time if `{sort_order}` is `ascending`, or the current time if `{sort_order}` is `descending`.
+  - `{sort_order}` is either `ascending` (the default) or `descending`.
+  - `{limit}` indicates roughly how many entries to return. It is expected that slightly more results will be returned, because if any entries exist for a timestamp, all entries for that timestamp are always fetched. It defaults to `200`.
```

Comment:
This statement is not very clear - "It is expected that slightly more results will be returned, because if any entries exist for a timestamp, all entries for that timestamp are always fetched". Are we saying that we could return more than 200 results in case of default limit? Basically what is "more results" relative to?

cc @rgcase

Reply:
If I understand it correctly: There is a bucket for each timestamp, and if your results include even a single item in a bucket, you'll instead get the whole bucket.

Reply:
Yes that's the idea. It's not expected that slightly more *will* be returned, it's just possible but honestly probably so unlikely that it'll never happen.

The reason for this is to make pagination work. If the user passed in a timestamp `t`, it'll return events with timestamp greater (or less) than `t`. But if we get unlucky and there are multiple events with the same timestamp, and the boundary lands between them, then the next attempt to fetch events using the latest timestamp from the results will end up skipping over the later of the events with the same timestamp.

We're talking about millisecond precision timestamps here, so in practice we'll probably never log two events with exactly the same timestamp, and even less likely, two events with the same timestamp in the same organization. Even if that were to happen, it would have to land exactly at the 200th (or whatever limit they pass) event in the response for this to even happen.

It's possible I drew too much attention to this scenario for how unlikely it is to happen, so feel free to soften it a bit or just mention it in passing that there's a remote possibility that the API will return more results than the `limit`.

Reply:
Got it, this is helpful. Yeah would be good to change this a bit to indicate that we may return more results than the limit if the timestamp for some of the events is exactly same, or something like that.